src/signTransaction.c

```diff
@@ -944,81 +930,16 @@ __noinline_due_to_stack__ void signTx_handleFinishAPDU(
-            uint8_t V[33];
-            uint8_t K[32];
-            int tries = 0;
-
-            // Loop until a candidate matching the canonical signature is found
-            // Taken from EOS app
-            // We use G_io_apdu_buffer to save memory (and also to minimize changes to EOS code)
-            // The code produces the signature right where we need it for the respons
-            explicit_bzero(G_io_apdu_buffer, SIZEOF(G_io_apdu_buffer));
-            for (;;) {
-                if (tries == 0) {
-                    rng_rfc6979(G_io_apdu_buffer + 100,
-                                hashBuf,
-                                privateKey.d,
-                                privateKey.d_len,
-                                SECP256K1_N,
-                                32,
-                                V,
-                                K);
-                } else {
-                    rng_rfc6979(G_io_apdu_buffer + 100, hashBuf, NULL, 0, SECP256K1_N, 32, V, K);
-                }
-                uint32_t infos;
-
-                size_t sig_len_ = 100;
-                CX_THROW(cx_ecdsa_sign_no_throw(&privateKey,
-                                                CX_NO_CANONICAL | CX_RND_PROVIDED | CX_LAST,
-                                                CX_SHA256,
-                                                hashBuf,
-                                                32,
-                                                G_io_apdu_buffer + 100,
-                                                &sig_len_,
-                                                &infos));
-                TRACE_BUFFER(G_io_apdu_buffer + 100, 100);
-
-                if ((infos & CX_ECCINFO_PARITY_ODD) != 0) {
-                    G_io_apdu_buffer[100] |= 0x01;
-                }
-                G_io_apdu_buffer[0] = 27 + 4 + (G_io_apdu_buffer[100] & 0x01);
-                ecdsa_der_to_sig(G_io_apdu_buffer + 100, G_io_apdu_buffer + 1);
-                TRACE_BUFFER(G_io_apdu_buffer, PUBKEY_LENGTH);
-
-                if (check_canonical(G_io_apdu_buffer + 1)) {
-                    TRACE("Try %d succesfull!", tries);
-                    break;
-                } else {
-                    TRACE("Try %d unsuccesfull!", tries);
-                    tries++;
-                }
-            }
-        }
-        FINALLY {
-            explicit_bzero(&privateKey, sizeof(privateKey));
-        }
+    uint16_t err =
+        signTransaction(&ctx->wittnessPath, hashBuf, G_io_apdu_buffer, SIZEOF(G_io_apdu_buffer));
+    if (err != SUCCESS) {
+        explicit_bzero(G_io_apdu_buffer, SIZEOF(G_io_apdu_buffer));
+        THROW(err);
     }
-    END_TRY;
 
     // We add hash to the response
-    TRACE("ecdsa_der_to_sig_result:");
     TRACE_BUFFER(G_io_apdu_buffer, PUBKEY_LENGTH);
-    memcpy(G_io_apdu_buffer + PUBKEY_LENGTH, hashBuf, SHA_256_SIZE);
+    memcpy(G_io_apdu_buffer + PUBKEY_LENGTH, hashBuf, SIZEOF(hashBuf));
```

Comment:
I don't see where we make sure that the current content of `G_io_apdu_buffer` is PUBKEY_LENGTH bytes,